kmod/src/xattr.c

```diff
@@ -864,18 +865,35 @@ static const char *full_name_hack(void *handler, const char *name, int len)
 {
 	return name - len;
 }
+#endif
 
-static int scoutfs_xattr_get_handler(struct dentry *dentry, const char *name,
-				     void *value, size_t size, int handler_flags)
+static int scoutfs_xattr_get_handler
+#ifdef KC_XATTR_STRUCT_XATTR_HANDLER
+		(const struct xattr_handler *handler, struct dentry *dentry,
+		 struct inode *inode, const char *name, void *value,
+		 size_t size)
+{
+#else
+		(struct dentry *dentry, const char *name,
+		 void *value, size_t size, int handler_flags)
 {
 	name = full_name_hack(NULL, name, handler_flags);
+#endif
 	return scoutfs_xattr_get(dentry, name, value, size);
 }
 
-static int scoutfs_xattr_set_handler(struct dentry *dentry, const char *name,
-				     const void *value, size_t size, int flags, int handler_flags)
+static int scoutfs_xattr_set_handler
+#ifdef KC_XATTR_STRUCT_XATTR_HANDLER
+		(const struct xattr_handler *handler, struct dentry *dentry,
+		 struct inode *inode, const char *name, const void *value,
+		 size_t size, int flags)
+{
+#else
+		(struct dentry *dentry, const char *name,
+		 const void *value, size_t size, int flags, int handler_flags)
 {
 	name = full_name_hack(NULL, name, handler_flags);
+#endif
 	return scoutfs_xattr_set(dentry, name, value, size, flags);
```

Comment:
Modern code still needs to call scoutfs_xattr_set() with the full xattr name, not just the name after the prefix.  full_name_hack() was doing that.  Modern code has to call xattr_full_name(handler, name).  If we're calling it directly instead of having ifdeffery in full_name_hack then we can drop the NULL handler argument to full_name_hack and update its comment.

Reply:
Clearly I didn't even read the comment o_O